Main.java

```diff
@@ -129,7 +135,7 @@ public void actionPerformed(ActionEvent e) {
       IDuelable secondPlayer = getDuelable(user2);
 
       // calls the duel method in the fight class.
-      int winner = Fight.duel(firstPlayer, secondPlayer);
+      int winner = Fight.duel(firstPlayer, secondPlayer, new Desert());
```

Comment:
I assume you want to use `currentEnvironment` here instead of hard coding `new Desert()`?
```suggestion
      int winner = Fight.duel(firstPlayer, secondPlayer, currentEnvironment);
```